### PR TITLE
Dma caching

### PR DESCRIPTION
**Pull request by alan-stokes**
adds bits and peices

Depends on https://github.com/SpiNNakerManchester/SpiNNMan/pull/233

tested by https://github.com/SpiNNakerManchester/sPyNNaker8/pull/508

### CODE REVIEW

**Review by rowleya — APPROVED** [review]
This is fine, but the integration tests are failing...

**Comment by alan-stokes** [comment]
done

File: c_common/front_end_common_lib/include/malloc_extras.h [hunk 1]
```diff
@@ -138,17 +138,26 @@ void *malloc_extras_sdram_malloc_wrapper(uint bytes);
...
 void *malloc_extras_malloc(uint bytes);
 
+//! \brief Allows a search of the 2 heaps available. (stolen SDRAM, DTCM)
+//! \note Commented out as this can cause stack overflow issues quickly.
+//!     If deemed safe, could be uncommented. That is the same to the
+//!     `#define` below at the end of the file
+//! \param[in] bytes: the number of bytes to allocate.
+//! \return: the address of the block of memory to utilise.
+void *malloc_extras_malloc_reverse(uint bytes);
+
```
Comment (Christian-B) [inline c_common/front_end_common_lib/include/malloc_extras.h:148]: Not commented out. But not safe comment not removed!

File: c_common/front_end_common_lib/src/malloc_extras.c [hunk 2]
```diff
@@ -658,3 +658,34 @@ void *malloc_extras_malloc(uint bytes) {
...
 }
+
+void *malloc_extras_malloc_reverse(uint bytes) {
+    if (safety) {
+        bytes = bytes + EXTRA_BYTES;
+    }
+
+    // try DTCM if allowed (not safe if overused, due to stack overflows)
+    int *p = NULL;
+    p = safe_sdram_malloc(bytes);
```
Comment (Christian-B) [inline c_common/front_end_common_lib/src/malloc_extras.c:669]: Comment "try DTCM" and  call "safe_sdram"... do not match up.

**Review by Christian-B — CHANGES_REQUESTED** [review]
A lot of thing changed to debug.  Is this to safe space or reduce the amount of messages?

File: c_common/front_end_common_lib/include/malloc_extras.h (hunk 1 above)
Comment (alan-stokes) [reply c_common/front_end_common_lib/include/malloc_extras.h:142]: fixed

File: c_common/front_end_common_lib/src/malloc_extras.c (hunk 2 above)
Comment (alan-stokes) [reply c_common/front_end_common_lib/src/malloc_extras.c:669]: fixed

**Comment by alan-stokes** [comment]
purely for space. if you turn them back to info. you lose even more binaries

**Comment by Christian-B** [comment]
This is so old I don't think it is worth saving.

Please reopen if it is.

Will take major effort to bring up to date with master.

**Pull request closed without merging**